### PR TITLE
Add optimized row count logic into Curator.

### PR DESCRIPTION
**Pull request by ryankwagner**
Check if the query is dimension only, dimension driven, includes row count, and has max rows of 1.  If so, remove ordering.

### CODE REVIEW

File: service/src/main/scala/com/yahoo/maha/service/curators/Curator.scala [hunk 1]
```diff
@@ -127,8 +127,35 @@ case class DefaultCurator(protected val requestModelValidator: CuratorRequestMod
...
       try {
-        val requestModelResult = requestModelResultTry.get
+        var requestModelResult = requestModelResultTry.get
         requestModelValidator.validate(mahaRequestContext, requestModelResult)
+
+        val isDimOnlyTotalRowQuery: Boolean =
+            requestModelResult.model.includeRowCount &&
+            requestModelResult.model.isDimDriven &&
+            requestModelResult.model.maxRows == 1 &&
+            requestModelResult.model.bestCandidates.isEmpty
```
Comment (pranavbhole) [inline service/src/main/scala/com/yahoo/maha/service/curators/Curator.scala]: it is assumed but its good to add dim candidates non empty

Comment (pranavbhole) [reply service/src/main/scala/com/yahoo/maha/service/curators/Curator.scala]: And also curators are empty

File: service/src/main/scala/com/yahoo/maha/service/curators/Curator.scala [hunk 2]
```diff
@@ -127,8 +127,36 @@ case class DefaultCurator(protected val requestModelValidator: CuratorRequestMod
       )
     } else {
       try {
-        val requestModelResult = requestModelResultTry.get
+        var requestModelResult = requestModelResultTry.get
         requestModelValidator.validate(mahaRequestContext, requestModelResult)
+
+        val isDimOnlyTotalRowQuery: Boolean =
```
Comment (patelh) [inline service/src/main/scala/com/yahoo/maha/service/curators/Curator.scala]: If we are changing the default curator, to special case a request which we only care about total rows, and the user is already passing in row count of 1, why can't they just pass empty ordering clause ?

Comment (ryankwagner) [reply service/src/main/scala/com/yahoo/maha/service/curators/Curator.scala]: You're right, the user was passing a problematic query & we shouldn't be updating policy to account for their actions here.  Steps have been taken on that end, and the query is fixed on their end.  Thanks for the input!  Going to close the related PRs for this.

**Pull request closed without merging**